pkg/envinfo/envinfo.go

```diff
@@ -0,0 +1,281 @@
+		Filename:          envInfoFile,
+		envinfoFileExists: true,
+		fs:                fs,
+	}
+
+	// if the config file doesn't exist then we dont worry about it and return
+	if _, err = e.fs.Stat(envInfoFile); os.IsNotExist(err) {
+		e.envinfoFileExists = false
+		return &e, nil
+	}
+
+	err = getFromFile(&e.EnvInfo, e.Filename)
+	if err != nil {
+		return nil, err
+	}
+	return &e, nil
+}
+
+func getFromFile(envinfo *EnvInfo, filename string) error {
+	pei := newProxyEnvInfo()
+
+	err := util.GetFromFile(&pei, filename)
+	if err != nil {
+		return err
+	}
+	envinfo.componentSettings = pei.ComponentSettings
+	return nil
+}
+
+// NewEnvInfo creates an empty LocalConfig struct with typeMeta populated
+func NewEnvInfo() EnvInfo {
+	return EnvInfo{}
+}
+
+// newProxyEnvInfo creates an empty ProxyEnvInfo struct with typeMeta populated
+func newProxyEnvInfo() proxyEnvInfo {
+	return proxyEnvInfo{}
+}
+
+// SetConfiguration sets the environment specific info like cluster host etc.
+func (esi *EnvSpecificInfo) SetConfiguration(parameter string, value interface{}) (err error) {
+	if parameter, ok := asLocallySupportedParameter(parameter); ok {
+		switch parameter {
+		case "url":
+			urlValue := value.(ConfigURL)
+			if esi.componentSettings.URL != nil {
+				*esi.componentSettings.URL = append(*esi.componentSettings.URL, urlValue)
+			} else {
+				esi.componentSettings.URL = &[]ConfigURL{urlValue}
+			}
+		}
+
+		return esi.writeToFile()
+	}
+	return errors.Errorf("unknown parameter :'%s' is not a parameter in local odo config", parameter)
+
+}
+
+// DeleteEnvDirIfEmpty Deletes the env directory if its empty
+func (esi *EnvSpecificInfo) DeleteEnvDirIfEmpty() error {
+	envDir := filepath.Dir(esi.Filename)
+	_, err := esi.fs.Stat(envDir)
+	if os.IsNotExist(err) {
+		// If the Env dir doesn't exist then we dont mind
+		return nil
+	} else if err != nil {
+		// Possible to not have permission to the dir
+		return err
+	}
+	f, err := esi.fs.Open(envDir)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.Readdir(1)
+
+	// If directory is empty we can remove it
+	if err == io.EOF {
+		glog.V(4).Info("Deleting the env directory as well because its empty")
+
+		return esi.fs.Remove(envDir)
+	}
+	return err
+}
+
+// DeleteEnvInfoFile deletes the envinfo.yaml file if it exists
+func (esi *EnvSpecificInfo) DeleteEnvInfoFile() error {
+	return util.DeletePath(esi.Filename)
+}
+
+// IsSet uses reflection to get the parameter from the envinfo struct, currently
+// it only searches the componentSettings
+func (esi *EnvSpecificInfo) IsSet(parameter string) bool {
+	return util.IsSet(esi.componentSettings, parameter)
+}
+
+// EnvInfoFileExists if the envinfo file exists or not
+func (esi *EnvSpecificInfo) EnvInfoFileExists() bool {
+	return esi.envinfoFileExists
+}
+
+// DeleteEnvSpecificInfo is used to delete environment specific info from local odo envinfo
+func (esi *EnvSpecificInfo) DeleteEnvSpecificInfo(parameter string) error {
```

Comment:
since for setting operation, the function above is called `SetConfiguration`. Then why not call this function `DeleteConfiguration`? Or maybe change the name of the set function to closer to this?

Reply:
updated